### PR TITLE
[MM-59899] Migrate tooltips of  'components/setting_picture.tsx' to WithTooltip

### PR DESCRIPTION
**Pull request by Camillarhi**
#### Summary

Migrate tooltips to use WithTooltip in 'components/setting_picture.tsx'  file
#### Ticket Link

Fixes https://github.com/mattermost/mattermost/issues/27782
Jira https://mattermost.atlassian.net/browse/MM-59899

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: webapp/channels/src/components/setting_picture.tsx [hunk 1]
```diff
@@ -231,16 +231,12 @@ export default class SettingPicture extends Component<Props, State> {
                     >
                         {imageElement}
                     </div>
-                    <OverlayTrigger
-                        delayShow={Constants.OVERLAY_TIME_DELAY}
+                    <WithTooltip
+                        id='removeIcon'
+                        title={<div aria-hidden={true}>
+                            {title}
+                        </div>}
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/setting_picture.tsx:238]: Lets just pass the title directly instead here

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]
Just one minute change

File: webapp/channels/src/components/setting_picture.tsx (hunk 1 above)
Comment (Camillarhi) [reply webapp/channels/src/components/setting_picture.tsx]: I have passed `title` directly

File: webapp/channels/src/components/setting_picture.tsx [hunk 2]
```diff
@@ -231,16 +231,10 @@ export default class SettingPicture extends Component<Props, State> {
                     >
                         {imageElement}
                     </div>
-                    <OverlayTrigger
-                        delayShow={Constants.OVERLAY_TIME_DELAY}
+                    <WithTooltip
+                        id='removeIcon'
+                        title={title ?? ''}
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/setting_picture.tsx:236]: is `title` not always defined? If its not defined we should disable them.

Comment (M-ZubairAhmed) [reply webapp/channels/src/components/setting_picture.tsx:236]: There is disabled prop we can utilize

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

File: webapp/channels/src/components/setting_picture.tsx (hunk 2 above)
Comment (Camillarhi) [reply webapp/channels/src/components/setting_picture.tsx:236]: > is `title` not always defined? If its not defined we should disable them.

yes, `title` is not always defined, the default value is undefined. When I disable the component when title is undefined, is it still alright to leave it as `title={title ?? ''}` as passing `title={title}` throws a typescript error

Comment (M-ZubairAhmed) [reply webapp/channels/src/components/setting_picture.tsx:236]: When an empty string is passed as title? How does the tooltip looks like

Comment (M-ZubairAhmed) [reply webapp/channels/src/components/setting_picture.tsx:236]: @Camillarhi can you check the above comment?

**Review by Rajat-Dabade — APPROVED** [review]
Apart from @M-ZubairAhmed comment everything else LGTM

File: webapp/channels/src/components/setting_picture.tsx (hunk 2 above)
Comment (Camillarhi) [reply webapp/channels/src/components/setting_picture.tsx:236]: image

It looks like this when an empty string is passed a title. I decided to utilize the disabled prop when the title is undefined

**Review by M-ZubairAhmed — APPROVED** [review]
This looks good @yasserfaraazkhan i did e2e test for this as well just to cross check if we had any related failures for this one

**Review by yasserfaraazkhan — APPROVED** [review]

**Pull request merged**